src/components/VmDetails/cards/NicsCard/NicEditor.js

```diff
@@ -0,0 +1,303 @@
+
+import {
+  Button,
+  Col,
+  ControlLabel,
+  FieldLevelHelp,
+  Form,
+  FormControl,
+  FormGroup,
+  Modal,
+  Radio,
+} from 'patternfly-react'
+import SelectBox from '../../../SelectBox'
+import ExpandCollapseSection from '../../../ExpandCollapseSection'
+import NicLinkStateIcon from './NicLinkStateIcon'
+
+import { msg } from '../../../../intl'
+import style from './style.css'
+
+const EMPTY_ID = ''
+
+const NIC_INTERFACES = [
+  {
+    id: 'rtl8139',
+    value: 'rtl8139',
+  },
+  {
+    id: 'e1000',
+    value: 'e1000',
+  },
+  {
+    id: 'virtio',
+    value: 'VirtIO',
+  },
+]
+const NIC_INTERFACE_DEFAULT = 'virtio'
+const NIC_INTERFACE_CANT_CHANGE = [ 'pci_passthrough' ]
+
+const LabelCol = ({ children, ...rest }) => {
+  return <Col componentClass={ControlLabel} {...rest}>
+    { children }
+  </Col>
+}
+LabelCol.propTypes = {
+  children: PropTypes.node.isRequired,
+}
+
+/**
+ * Collect required information required to create a new NIC for a VM.
+ *
+ * Required data:
+ *   - Name (auto name the next number higher?)
+ *   - VNIC Profile (which includes the network) <-- needs to be on same DC and the VM
+ *
+ * Optional data:
+ *   - Network Type (dual mode virtio and rtl8139, virtio, rtl8139, e1000)
+ *     - NOTE: can only be changed if VM is down or Nic (card) is unplugged
+ *   - Link State (Up = connected, Down = cable unplugged)
+ */
+class NicEditor extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      showModal: false,
+
+      id: undefined,
+      name: props.nextNicName,
+      vnicProfileId: EMPTY_ID,
+      interface: NIC_INTERFACE_DEFAULT,
+      linked: true,
+    }
+
+    this.open = this.open.bind(this)
+    this.close = this.close.bind(this)
+    this.composeNic = this.composeNic.bind(this)
+    this.changeName = this.changeName.bind(this)
+    this.changeVnicProfile = this.changeVnicProfile.bind(this)
+    this.changeInterface = this.changeInterface.bind(this)
+    this.changeLinked = this.changeLinked.bind(this)
+    this.handleSave = this.handleSave.bind(this)
+  }
+
+  open (e) {
+    if (e && e.preventDefault) e.preventDefault()
+
+    const nic = this.props.nic
+    if (nic) {
+      this.setState({
+        showModal: true,
+
+        id: nic.id,
+        name: nic.name,
+        vnicProfileId: nic.vnicProfile.id,
+        interface: nic.interface,
+        linked: nic.linked,
+      })
+    } else {
+      this.setState({
+        showModal: true,
+
+        id: undefined,
+        name: this.props.nextNicName,
+        vnicProfileId: EMPTY_ID,
+        interface: 'virtio',
+        linked: true,
+      })
+    }
+  }
+
+  close () {
+    this.setState({ showModal: false })
+  }
+
+  composeNic () {
+    const nic = {
+      id: this.state.id,
+      name: this.state.name,
+      vnicProfile: {
+        id: this.state.vnicProfileId,
+      },
+      interface: this.state.interface,
+      linked: this.state.linked,
+    }
+
+    console.log('composed nic', nic)
```

Comment:
Did you forgot to remove this?

Reply:
FWIW - the log only outputs when `AppConfiguration.debug` is `true`.  It would have no effect in prod deployed code.